### ISSUE TITLE
Game theme / style (art)

### ISSUE THREAD

**Issue by flip40** [DemoProductions/shmup#6]
What should the overall theme or style be? Space shooter? Plane / tank shooter? Something else wacky?

Also any opinions on sprite style? We could go a bit more pixel-y, arcadey (not sure what to call that as an art style), a cell shaded style like last time, or whatever else you guys might come up with for styling it.

**Comment by ghost** [comment]
A pixel-y styled game sounds good to me. And maybe something not in space

**Comment by stannardtyler** [comment]
i dig the pixel art, I've never done anything hugt but I do have experience. Whats double awesome is that we can all easily make out own art ^.^ it ez and we can all share ideas! Now we can do two things to help brainstorm, either a basic like army aircraft type of ship and theme that can go anywhere. Or OR! like a different feel like a the player playin as a pterodactyl shooting fire blasts or eggs or something , mayb incorporate melee attacks of some sort. But it dont matter jus two ways to broadly help to narrow down the main theme. Thoughts?

**Comment by stannardtyler** [comment]
I can draw some crazy shit that is thought about and that can give us an idea how it would look after we start it up with placeholders.

**Comment by flip40** [comment]
Well I am more a fan of conventional, but then, that has always been done before. If we do something less conventional though we have to make sure it fits with the common upgrade system used in these type of games (I don't think a pterodactyl shooting silly amounts of eggs would really work, maybe a pterodactyl with laser cannons thought).

And Adrian, you say not in space, but does that necessarily mean not space-age (per se)? I mean, would you like to have it be more "aircraft" style as opposed to "spacecraft"? I can agree with not having space be the background of the setting either way though.

Either way, I think the armament would be easiest as lasers, missiles and maybe bombs, they fit the idea of getting upgrade pickups within the levels nicely.

**Comment by stannardtyler** [comment]
i had a game on the sega that was totally badass mechanically and style wise . I'll find a link of the game tomorw and share if that's a certain style we wana go for.

**Comment by stannardtyler** [comment]
lightening force is the game, i know this video is a shitty review but ignore his talk this game is badass. And the pixelated look i think should kind of look like this. Yes it is a space ship but not in space like Adrain wants too avoid. But we can be influenced off this and other games. Show me some influences so I can create some ART! :D 
 https://www.youtube.com/watch?v=UMB-OAodJ4w

**Comment by stannardtyler** [comment]
ok the review isnt that shitty, I've played plenty of shooters but he does talk about certain mechanics and events that may help us with our game.

**Comment by flip40** [comment]
woops, didn't mean to close.

**Comment by ghost** [comment]
Disregard my "not being in space" comment haha. I was thinking space as in Space Age. 

I still have not seen the videos mentioned above and on Facebook. I will leave a comment and my idea on this later tonight

**Comment by ghost** [comment]
spaceship that looks kind of like a house 

![spaceship1](https://cloud.githubusercontent.com/assets/10818718/16401934/6dc3e82e-3c9d-11e6-9bba-59f76402fe05.png)

**Comment by flip40** [comment]
I say around 64x64 for most things, possibly even bigger for others. Few things should be smaller (mostly things we don't need to intentionally be shooting).

![size comparison](https://cloud.githubusercontent.com/assets/10754057/16882936/6579990c-4a76-11e6-8ca4-565df35b797c.png)

**Comment by stannardtyler** [comment]
This was something I did on the go, a simple space ship top down style, I can't recall if we wanted to stick with the space theme or somethin with more character. Like the art style in the tutorials game. I can update with some sketches soon. 

The img is just a screen shot, didn't think to send a png, as we are doing parallax scrolling rather than top down.

Checking out the unity file, I see we have it set up for horizontal, so I can redesign a horizontal viewed 
![untitled](https://cloud.githubusercontent.com/assets/15371385/16889660/db48e1ee-4a9c-11e6-9c39-c425408f4e6f.png)

I've taken the ship above and began sketchin some concepts up. I'll prob for now, focus on making a basic ship, prob 32x32 for now, so art isn't so detailed that others can't jump in and add assets. I gona make sure the ship is the same size for animations so the hitbox for the player is constant for gameplay. Here's the sketch with other ships along as well, maybe ideas for enemies and such. We can always switch up the art style as we go.

![img_2094](https://cloud.githubusercontent.com/assets/15371385/16890374/9899fdbe-4aa2-11e6-869c-9ae9c92cc103.JPG)

The main ship on the left is what I'll mock up for now in our game.

**Comment by stannardtyler** [comment]
Also curious, so much focus I've been doing is the main player, but what should our enemies look like? I see we have several using different movement tactics, so curious if we should make their tactics noticeable via the look of a certain enemy ship/alien.

**Comment by stannardtyler** [comment]
Here's a prototype for our ship, 32x32 is tight to work with, I can give 64x64 a shot. Or jus upscale it.
![ship7](https://cloud.githubusercontent.com/assets/15371385/16891837/868d4352-4ab5-11e6-801a-3c52c93c9d77.png)

**Comment by flip40** [comment]
One of my reasons for commenting 64x64 was also just for the size it will be on the screen, and how difficult it may be to hit. Of course, we could potentially just scale it (though we should attempt to make sure the art all is drawn to the same scale so they all use the same Unity scale... if that makes sense).

But yeah, that player ship definitely looks awesome :D Having ships that have an appearance fitting for their tactics (both movement and weaponry) seems like a good idea, makes design of art and mechanics become a bit synergistic.

Also feel free to create designs that might use movement or weaponry tactics we have not created yet. While currently our movement is only linear or sin / cos based, we do plan to add other things, so if you have ideas let us know.

**Comment by stannardtyler** [comment]
ok ok , I'll make some time to make some enemy ships as well, basic at first for placeholders and we can always expand later.

**Comment by ghost** [comment]
From Facebook:

![ship2](https://cloud.githubusercontent.com/assets/10818718/17272341/bf8ac1f4-5648-11e6-82f1-1901b8748db1.png)

![ship3](https://cloud.githubusercontent.com/assets/10818718/17272343/c5d46380-5648-11e6-811b-e8e164c110ad.png)

**Comment by stannardtyler** [comment]
OK here is the latest ships and a projectile sprite sheet, jus a basic projectile, more ideas on bullets in that discussion post. 
![ship4](https://cloud.githubusercontent.com/assets/15371385/17272503/98b8dcce-564c-11e6-9b2d-331820d54d84.png)
![shipv](https://cloud.githubusercontent.com/assets/15371385/17272504/9b73d6f8-564c-11e6-98bf-c88e7830e391.png)

the blue ship was designed as a purpose to fly low beneath the player, most likely grazing the bottom of the screen, stationary or moving slowly, or follow? The Blue pillars are designed to move later on, going up and down shooting bullets, or mines? below or above the player down on them possibly as well. For now they can just shot since only one projectile for now, unless someone wants to add some art! 

The pink ship I designed like a mini nemesis, this ships role would be that of a "homing ship". Those enemies that fly quick in and follow you, and begin to shoot you for abit, until some time or is killed. Tried to make it a sexier looking ship, unique to show it's threat. It has 3 engines and a bigger gun as well that will effect it's later animation and attacks/movements possibly. Also 4 wings like an X-wing lol.

Lastly the two ships before, besides the players red ship, the green and yellow can be used as basic units commonly seen. Green can zig zag, while yellow is a straight shot maybe? 

Lastly this is all the art I have created so far, curious as if this art style is fine for now, we seem to be going towards a classic side scrolling space shooter with waves of enemies and bosses. I feel we should hone that and make it fun, then maybe add elements of innovation if want!

Also, lmk if the art and feel is going ok, because I can always switch up styles, add more detail and make this a polished game. Just that will take time, but I can do this like a second job in a sense.

**Comment by stannardtyler** [comment]
sorry for wall of text, ask questions if need, in rush to work so typing fast.

**Comment by flip40** [comment]
Will get these added asap. I really like how the ship designs fit for an attack pattern, and at least gives me ideas on attack patterns to work on. Not sure how to interpret the bullet sheet, are they separate bullets or an animation? If the latter, what is the expected animation? Constant flashing, just one time, etc.

I think the ship styles are good. Only comment is that they don't feel factional (part of a group), if that makes sense. Some ideas:
- A color pattern or scheme that makes them feel like they are together. Of course, too much similarity and everything just looks the same.
  - A simple thing towards this could be differing engine colors for different groups perhaps?
  - How about bullet color? Though that is only visible while shooting, could still add though. Cool colors for friendly and warm colors for enemies?
- Ship style. Probably much more difficult to actually implement but this is all just speculation really anyways. Some things that come to mind (if a bit vague) are terms like "angular (sharp!), square (blocky!), round (bulbous!.. sorry...)" to describe the designs that make each group's ships unique.

Of course, perhaps that needs more speculation on the plans for the game, but I think at least something that groups "enemy" ships apart from "friendly" ships would be a good idea (which vies more towards color or pattern as opposed to the whole ship style).

Anyways just some thoughts, what we have so far is definitely great though :D

(related to the thoughts on having ships be factional: https://github.com/DemoProductions/shmup/wiki/Story)

**Comment by stannardtyler** [comment]
Well noted! I'll take these comments into consideration. Make the enemy groups unique yet the whole enemy force as a whole.

**Comment by stannardtyler** [comment]
Also yes the bullets are like an animation flashing, something quick and ez for a player to see. If it's too small, increasing size shouldn't hurt.

**Comment by stannardtyler** [comment]
Here is the 2D art I have completed so far for the game. The last I'll most likely focus on further animations, such as the power ups, and possible death, or low health visuals. Also a possible end boss and backgrounds.

Boss maybe something I'll create or Jamahl Reynolds, as it might be cool and interesting to have a giant 3D model battling as a end boss. Maybe the 3D vs 2D game style I mentioned on earlier FB posts. 

Concepts and ideas on background will come soon.

Also the retouches for ships, i.e color palette, look, can come later, but hopefully what is provided can get a sense of a first level. For now since everything is flat I suggest we focus on a level where it's a typical layout of going to the right and enemies appear and go left off screen. 

Enemies and obstacles such as asteroids can come from any direction honestly, but without any turning animations it will most likely look abit awkward visually lol, definitely a paper Mario look. 

The asteroids are there for non shooting obstacles to look out for, the animations and art are kinda tied in with the images. So options such as the large rock is shot, splits into the normal rocks, then to the small. I think all that can be done in unity if I remember like cutting the images up and making your own timeline animation, but myself not really having hands on i can't say nor know. If not, I can make some of the events and animations as gifs or sprite sheets in photoshop if it helps! The small rocks are more so as debris that doesn't hurt the player. Maybe fun if they could bounce off your ship! 

Also the power ups, although with no shield animation, can spawn from the asteroids. Maybe the bigger the rock the more chance there is to get one. Speaking of that, I recalled we have a hp bar idea floating around. Let me know have we decided to use a hp bar and lives, or a system of 3 hits, making the ship look worse and after the 3 hits you die with lives as well, or lastly the traditional 1 hit dead, with the shield helping and a few lives?
[Space Game Art 8.17.zip](https://github.com/DemoProductions/shmup/files/423531/Space.Game.Art.8.17.zip)

I'll also be available to try any prototype builds that have came around recently to test mechanics and see the art in action. The last version of the game I downloaded just had the sky background, along with some shapes. 

Goodluck reading all this.

[Uploading Space Game Art 8.17.zip…]()

![nemesis](https://cloud.githubusercontent.com/assets/15371385/17751348/a154ed40-647b-11e6-9414-8fa636cb38e7.png)
![new bullet](https://cloud.githubusercontent.com/assets/15371385/17751349/a1555c8a-647b-11e6-8e7a-6345d1352592.png)
![og bullet](https://cloud.githubusercontent.com/assets/15371385/17751351/a155bb1c-647b-11e6-9c2b-9ff9cc4a8ca8.png)
![bottom-top](https://cloud.githubusercontent.com/assets/15371385/17751350/a1557198-647b-11e6-96df-03a4f2861f9e.png)
![diagonal](https://cloud.githubusercontent.com/assets/15371385/17751352/a1565590-647b-11e6-8c6f-c7ab93323982.png)
![mid-straight](https://cloud.githubusercontent.com/assets/15371385/17751353/a15710de-647b-11e6-98fb-bc26fb0e198a.png)

![normal rocks](https://cloud.githubusercontent.com/assets/15371385/17751364/a9eea40a-647b-11e6-8598-902058edd99b.png)
![small rocks](https://cloud.githubusercontent.com/assets/15371385/17751362/a9d1821c-647b-11e6-9815-38676621c229.png)
![big rock](https://cloud.githubusercontent.com/assets/15371385/17751363/a9d653d2-647b-11e6-8cea-536897f22726.png)

![player](https://cloud.githubusercontent.com/assets/15371385/17751368/aee4f3f6-647b-11e6-9b56-0bc06613fdde.png)
![player-bullet](https://cloud.githubusercontent.com/assets/15371385/17751369/aee8d32c-647b-11e6-88f7-d36be5a95f49.png)
![player-up bullet](https://cloud.githubusercontent.com/assets/15371385/17751370/aeeb6e84-647b-11e6-9048-616409e3ddcf.png)

![damage](https://cloud.githubusercontent.com/assets/15371385/17751374/b378ce4c-647b-11e6-8b41-180150cdd2d8.png)
![shield](https://cloud.githubusercontent.com/assets/15371385/17751373/b376de7a-647b-11e6-93f3-c189004419b3.png)

**Comment by flip40** [comment]
Some feedback, would you mind keeping everything in the same rotation? I.e. all the enemy sprites are backwards relative to the player. For simplicity, everything should always face right. Reason for this is due to some abstraction in the code and game objects that makes rotation and such convenient. For example, we can tell a player and an enemy to simply "go forward" in an abstract way if they face the same direction (because forward is to the right for the art), and then simply rotate the game object to face the other way (which makes ITS forward now actually be left). As is I now have to either rotate the sprites or explicitly tell enemies to go the opposite direction of the player, which also breaks other things (for example, they all shoot backwards now too).

I will mirror them on my end for now.

Everything does look great though!

**Comment by stannardtyler** [comment]
ok i'll make sure to keep the incoming art all the same direction! lol silly me. Ok next thing on my list is simple backgrounds I think to set the mood. That video mash up I did and posted to our facebook is something along the lines of the final look I think we should aim for. Alot the the shooters we were influenced from earlier had big art to show the detail and action. So something on a scale of Super R Type or metal slug with size proportion, scale, and game view size. Thoughts?

**Comment by stannardtyler** [comment]
Here's an updated sprite sheet of the art that has been done with a few tweaks. Noticable additions would be the parralax backgrounds. They can be used to be placeholders as I'll use their outlines to add more details. Thinking maybe a first level on a planets surface rather than in space. 
Also the enemy bullet sprites only have one for now, but you can look at the ships to design unique ways they can shoot, I think I explain my designs for them in another post.
Also lastly since our player ship is the main character for now, I added an flying animation with the fire pulsing in the back. Hope this is something towards the right direction. I'm gonna focus on getting enough art to complete the first level, and leave animation for fun details towards then end of the winter break for myself lol.

[Sprites 1.5.17.zip](https://github.com/DemoProductions/shmup/files/688696/Sprites.1.5.17.zip)

Also I heard talk of a boss character, have any designs been made? I'm curious what style I should make the boss. If no ideas I'll sketch some up tonight.

edit: while testing the files some look blurry, lmk if the pngs upload clearly on unity.

**Comment by flip40** [comment]
Awesome, I'll add in the parallax asap. I believe we already have the code/tech in place for those, just need to set em up. Ditto on adding the animation for the player ship.

Yeah, I recall discussion of the enemy designs. I believe the movement for most of those are already set up. Need to get them shooting properly though (and actually put them in some waves, and those waves into the level! Currently they are all green guys except for the boss).

Speaking of the boss, depends on what you mean exactly by "have any designs been made". There has been work done on designing some boss AI, however we currently just used the "nemesis" (purple) ship for him. Personally I think the Boss AI should always be a bit (or significantly) larger than normal. They'll take a lot of hits, and we actually _want_ them to be hit. The player should find difficulty in staying alive and dodging through bullets to be in a position to hit the  boss, however it should not be hard to hit him due to being small and moving around rapidly.

I would prefer to use a different boss (image / size / design) and reuse the pink ship design into another AI type. Possibly a "mini boss" of sorts due to having some added intelligence over the other AI, but otherwise normal. I'll start a thread on boss states / stages and we can work on that (feel free to start sketching up ideas!). I don't mind if we replace some of the current boss AI if we want to hit a specific artistic design or whatever. We can always reuse the current states on another boss, or even as "mini boss" states if we want.

And lastly, mind lmk which png's you think are blurry? I took a quick look and _most_ are using point filtering (i.e. no filter) however some things we haven't stuck into the game yet (but were added to the unity project) still use bi-linear filtering (it's the default) so that is likely what you are seeing, but it would be good to double check. I'll make sure everything uses point filter when we get the new sprite sheet in.

Edit: New issue for boss design discussion: #145

**Comment by flip40** [comment]
By the way, the sprite sheet has a white background, normally needs to be transparent. It would also be preferable if sprites were separated into their own sheets based on object. If all sprites are in a single sheet, I'll have to replace all sprites to fit the new sheet.

To clarify, all four player sprites would be in one sprite sheet, and all green bullets would be one sprite sheet, the single purple ship sprite would be in its own sheet and so on.

I'll change the sprites on my end for now so I can get them into Unity.

**Comment by stannardtyler** [comment]
Like Homer Simpson would say, "Doh!"
Been so long I jumped the gun and forgot the basics lol. I'll send them again the right way tomorrow. Also been sketching up some ideas for the look of the first level boss so I'll share my sketches tomorrow as well.

**Comment by stannardtyler** [comment]
Hey jus updating things with a new clean and updated sprite sheets for new and original assets.
Since you've got the previous ships and enemies, I chose not to resend them out here but I will have them for my own records.
But the backgrounds, bullets and such I resent, plus new animations : explosion and shield animation with power up.

We can use this explosion for everything for now until I make more unique ones, i.e. for the player or asteroid or even other ships!

Next stages of art I'm focusing on the boss, the "skybox" which is jus the space in the background that I haven't done, maybe some planets and galaxies to fully immerse which would be the real far background. So we have like a total of five 640x240 parallax backgrounds, lmk how that goes or I'll check it out on the newest version to see if we even need that much, it might add a lot of depth to our game!

Other than that good luck, inform me with high issues as I will be checking the forums daily for priorities in art assets. 

Also I will be looking towards title cards and such, nothing crazy on the UI side but at least a title screen possibly, I always feel that makes the game more real lol. ttyl

[sprites1.9.17.zip](https://github.com/DemoProductions/shmup/files/694799/sprites1.9.17.zip)

**Comment by flip40** [comment]
Sounds awesome! Sorry for the delay, I saw your update yesterday (or two days ago?) with Boss stuff, but was busy over the weekend. I'll be taking a look through all of this (and the boss stuff) and adding as much as I can when I get on the train after work today.

**Comment by stannardtyler** [comment]
no sweat, I just finally gettin in the swing of things myself. Had a goal to start pumping out assets but came over a wicked cold that I got after the new year. Just now I'm feeling better and I got some adrenaline from making some art and looking up reference material!

Gonna start playing those games we were influenced by and check out the mechanics and feel of the game while prototyping our own, excited for the updates and applaud the work that has been done on yours guys end.

**Comment by flip40** [comment]
Some more recommendations on future editions, bullets need a bit more spacing left to right between each other. Unity has a convenient feature for auto-slicing these, but I guess it thought the first two on each sheet were a single sprite. No need to get me a new set, I already manually selected them, but it would be nice for future.

The explosion and shield effect look great. I doubt I'll get everything added in tonight, but should be able to have all your new assets in by tomorrow night (that's the plan anyways).

**Comment by flip40** [comment]
I think the background / foreground (parallax) will need to be bigger. I ended up having to scale them up by 3.5 to be wider than the screen, and the pixels are very noticeable when they are over 3 times larger than normal :P Unfortunately I doubt there is an easy way to expand your png's without causing the same pixelation issue (but then again, photo editors are pretty advanced these days, maybe they know how to zoom smartly to avoid pixelation? Might be a filter or something that does that...)

I need to make some code changes to properly add in all 4 backgrounds, but I placed them in the scene with the proper scrolling attached and it looks great though (pixelation aside). Feels so much better than our old stand-in blue sky and clouds background!

**Comment by stannardtyler** [comment]
ok not take on bullets, I felt that way as well when I exported, was thinking it was too tight. But glad only 3 frames lol I take note. 

Also for the resize of  background, here is a new sprites that are to size that was need, rather than the 640x240 I multiplied it by 3.5 as requested to keep the hard edges. Here are the 4 backgrounds resized to 2240x840. Lmk how this goes once coding is finished. I'm thinking to start on the boss sprite since we can use all the boss concepts for other future stages!

[sprites1.10.17.zip](https://github.com/DemoProductions/shmup/files/697402/sprites1.10.17.zip)

**Comment by flip40** [comment]
I might have been unclear in my message, but I would prefer the background to be 3.5 but without the hard edges. In other words, we want the edges to not be blurred, however we don't want it to appear overly pixelated in relation to the rest of the art either. Unless I am mistaken and you prefer that for stylistic reasons, I think this would be better.

Here is an example of what I am talking about. From top left, to bottom right:
1) original
2) purely scaled 3x (don't want)
3) scaled 3x and "smoothed" (do want)

![example](https://cloud.githubusercontent.com/assets/10754057/21827873/58d979f2-d742-11e6-8e51-3cee9d4f1dd0.png)

I don't know for sure how much this can be modified, but I figure for every multiplication, we get a number of extra pixels that can make the transition look less jaggedy. I don't have time to make another example while at work, but for example directly related to mountains and jaggedy:
Every column of pixels up eventually moves one pixel over at normal resolution. Scaling 3x causes every 3 columns to reach the same height, before moving over 3 more columns. Instead, we would want this to smooth itself back to the 1 column pattern on the original. This could be done manually by, for each column of 3 pixels that reach the same height, slope them appropriately (i.e. move one side up a number of pixels, and the other side down a number of pixels, middle column stays). Not that I recommend making that change manually, but just trying to convey the idea of what I mean by needing more pixels / less jaggedy-ness by removing the blockiness we get from pure scaling.

Of course, I don't want to take your time away from doing fun things either :P this is relatively low priority compared to something like getting bosses done or what not. Heck, if I have time this is something I could probably do myself while you work on the boss designs (I am assuming you would be more interested in working on something like the boss design than redoing background mountains :P).

Of course, an interesting note on top of all this is that most things are already being scaled by 2x to even 5x (bullets) because they felt too small in terms of size for hit boxes. I just noted that my brother mentioned that that backgrounds were notably pixelated compared to the rest of the art, which isn't nearly as noticeable despite scaling.

**Comment by flip40** [comment]
Also I am going to be a bit delayed on getting all the sprites in it seems. I am having odd issues with Unity that I am unsure how to fix at the moment. Everything with animation editing in Unity lately seems way more clunky than it was when I worked on them for the Fighter game, and on top of that, a number of images in the animation seems to get modified by Unity for some reason causing a noticeable flicker (for example, near the front end of the player ship, the window and gun get colors bleeding into each other on particular frames, despite close observation showing that your png's for each ship are identical other than the engine effect).

So on one hand, why are Unity animations suddenly such a pain to work with, and why is Unity changing our sprites. Fun times.

**Comment by stannardtyler** [comment]
Silly me I getcha! make it smoother, since it's bigger lol. Yeah no sweat here's some assets that I think should fit the bill. They are around that big resolution size we are looking for with added pixels to give off a  more smooth slope with the hills.

[background1.10.17.zip](https://github.com/DemoProductions/shmup/files/698247/background1.10.17.zip)

It's a bit choppy of course, but will give us the idea of the look. If of course I understood this time haha. 

But, I also plan on adding some more details later to the foreground and some touches to the background to add some atmosphere. Even tho we are set in a space sci-fi theme we can avoid the cliche of a simple space level and have cool immersive planets we can conquer. Maybe a mega-man type stage system.

**Comment by flip40** [comment]
Certainly, I was actually thinking that the fact that there was just a solid single color for the background was probably making the pixels far more noticeable than they might otherwise be, and that adding details later might help :P regardless, you got what I was looking for on the background, I'll be getting those imported asap (will work on this on the way home again to try and get everything in, figured out some of my issues I was having previously).

I also figured out what was wrong with the player ship sprites, and for documentation's sake:
![screen shot 2017-01-12 at 3 16 44 pm](https://cloud.githubusercontent.com/assets/10754057/21912399/86320196-d8da-11e6-8a51-ed8597eb0136.png)
On the sprite, this needs to be overridden and set to a true or non-compressed color format (RGBA 32 bit in this case). Previously it was using some compressed format, causing very noticeable discoloration due to our sprites being so small. The compression would be unnoticeable on larger sprites, but in our case the change is very noticeable. We just need to remember to set this when we import new sprites so they look as good as they were sent to us :P

**Comment by flip40** [comment]
Hey Tyler, are the white pixels here supposed to be part of the bullet sprite? Assuming for now they aren't and am removing them, but wanted to confirm.

![screen shot 2017-01-13 at 2 51 07 pm](https://cloud.githubusercontent.com/assets/10754057/21948391/db37bff8-d99f-11e6-8d25-78c0b965b0b9.png)

Oh, and an update, so far I've got the backgrounds, explosions effects, player animation, and bullet animations all in and working. Powerup's will take a bit more time as I don't even have code yet to implement those, but they'll be in Soon™.

**Comment by stannardtyler** [comment]
Yea it doesn't matter too much, I think with the design I was trying to make a spinning round shape, but with the limited pixel I kept coming with a square lol. The mission was to make the bullets more noticeable as well, so we'll test the without for now and c how it looks in the end :) and if needed re-adjust them.

![boss_concept](https://cloud.githubusercontent.com/assets/15371385/21953032/2e39135e-d9e2-11e6-88e9-df488c48618f.png)

also working on boss I'm digging his look, you see I have multiple personalities while designing the look and such lol but hope you guys like the look :) open to any suggestions as well! :D

**Comment by flip40** [comment]
The front end reminds me of a Metroid. Definitely got that alien look going, and it looks great!

I'll use the bullet image without the white for now. Without anti-aliasing you can't do perfect circles at low res (perhaps not even with them), however so long as you keep a plus like shape you mimic it well enough for most people to interpret.

![image](https://cloud.githubusercontent.com/assets/10754057/21953213/bd4bad06-d9e5-11e6-9369-89e7a6248be7.png)

Also pixel circle chart for reference, but without the white pixels, you have basically a 4x4 circle in this style (like the 5x5 in the image).

**Comment by flip40** [comment]
All previous assets (minus asteroids) have been added and a Pull Request was made. I'll merge it tomorrow so it will all be in the master branch (i.e. you can easily grab the code and try it out yourself tomorrow!). For now though, I made a quick video so you can see it. My Macbook is clearly not a video recording machine, however, so the frame rate of the video is quite low. Should still get an idea of all your sprites being there though, and once I merge the PR tomorrow you can get it yourself from the master branch.

https://youtu.be/pWNoq0jr8b0

**Comment by stannardtyler** [comment]
I loved the demo of whats become lol , the nemesis mechanics at the end were pretty badass! haha
Wanted to chime and say like the work being done so far. Was curious how you feel about playing the game did the sizes of the sprites feel to small, should we zoom in abit? Also I think once we get other projectiles goin, i.e. a following missile or something I think that will greatly enhance the dynamics of the game.



![bossdemo1 16 17](https://cloud.githubusercontent.com/assets/15371385/22008087/b53fcdf6-dc2c-11e6-9057-248ffeb9d40b.gif)

So far I got he boss sprite completed, and now working on animations and attack pattern. Planning on two main attacks. One is when the little mouth open it sprays out projectiles randomly which the player will have to dodge. The other attack is to be a laser attack from the central hole thats on top of the boss.

No plans how it should take dmg, i.e. when it's cooling down from laser or just a big life bar that takes damage anywhere and anytime. Either sounds good to me. I was also planning to animate the bigger talons on the ship to give it a living creature vibe. In photoshop the pixels can't be rotated as well so curious if we can leave the talons as separate sprites and put them under the ship and animate them as if they were opening and closing in unity.  If not might hold that off for later polishing.

**Comment by flip40** [comment]
Sprite sizes seem fine at the moment (noting that we currently have a lot of sprites scaled to 2x, mostly ships and bullets). The game does feel a bit off, though. Currently I think this is largely due to a need to vary the bullet speed and movement speed of both player and enemy objects. It is hard to hit things at times, but I feel that is due more to how long the travel time of the bullet is than sprite sizes. We might need to change damage values as well. Each wave is also extremely boring... A combination of higher enemy density, though easier to kill, would help. Of course, I definitely welcome second opinions on sprite sizes and object speeds or whatever, that is just my take on what I think needs improvement. A lot of those will take time to work out due to interconnected mechanics though (more enemies would mean more bullets, so that implies making them easier to kill and/or shooting slower, and/or improving the player RoF / damage... etc).

We do actually have tacking missiles, just currently don't use them (using stand in sprites from the bullet sheet I got a while ago).

Boss sprite is looking awesome! We can definitely take the bigger talons (mandibles?) as separate sprites and animate them manually if you'd like that. No idea how exactly that will look, assuming your rotation issue is in regards to pixelation, but we can definitely do sprite rotation like that pretty easily with AICommands `First(Rotate(-30, 3)).Then(Rotate(30, 3)).End()` or something like that. Pretty easy adding a manual script for that too if we want something more defined or random or whatever.

The attacks sound good. Something that might save you some time for animation, have the engine animation be a separate set of sprites (same sprite sheet though). That way we can attach the engines effect as a separate object like the talons but have them work on a separate animation timer using the one repeating engine animation you have. The idea with this is so you don't have to worry about the engine pattern when doing effects for shooting or whatever else. For example, if you animate the little mouth in the center opening, it might need to pause for a few seconds (or worse, an indeterminate time frame if we want to be able to adjust it in the editor). In that case, you need a number of open mouth frames that keep the engine effect going... etc. Of course, we could also just pause the effect and it would be fine, but ya know. Whatever works. At the very least it might help so you don't need to draw the engine loop more than the one time with every animation you might add for the boss :P

TL;DR: Separating the parts of ship animation (such as the engine) and putting these pieces back together in Unity might make the overall animation of a ship like that require less redrawing and art time as well as be more dynamic as each ship piece could animate separately from the others.

Edit: Final note, the changes I mentioned and was visible in the demo vid is all in master now, if you feel like running it yourself and giving it a test.

Edit2: I just realized that the alien ship has a translucent skin, which is a cool (and quite appropriately hits the "living creature vibe" you mentioned). You probably are aware, but if you do the separate talon sprites, make sure you leave those skin areas as partially transparent (er, 50% alpha or whatever it is) so the separate sprites still have that effect :D (which maybe the already are, depending on how you decided to put the parts together anyways)
